sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -161,6 +175,43 @@ export class ApplicationController extends BaseController {
     const student = await this.studentService.getStudentByUserId(
       userToken.userId,
     );
+    /** Validation for application overlapping dates or Pending PIR.
+     * This validation can be disabled by setting BYPASS_VALIDATIONS_AT_LOCAL to true in .env file.
+     * */
+
+    if (!this.config.bypassValidationsAtLocal) {
+      const existingOverlapApplication =
+        await this.applicationService.validatePIRAndDateOverlap(
```

Comment:
I would say that, usually the overlap will not happen and the three queries will be executed, so I believe that the three queries can be executed in parallel, what also eliminates the need of creating the method _throwExceptionForPIRDateOverlap_ because after the parallel execution returns we can check if any query return something we can display the error.

Reply:
All the validation resolved as single promise as discussed.